plugins/zify/zify.ml

```diff
@@ -0,0 +1,1095 @@
+(************************************************************************)
+(*         *   The Coq Proof Assistant / The Coq Development Team       *)
+(*  v      *   INRIA, CNRS and contributors - Copyright 1999-2019       *)
+(* <O___,, *       (see CREDITS file for the list of authors)           *)
+(*   \VV/  **************************************************************)
+(*    //   *    This file is distributed under the terms of the         *)
+(*         *     GNU Lesser General Public License Version 2.1          *)
+(*         *     (see LICENSE file for the text of the license)         *)
+(************************************************************************)
+
+open Constr
+open Names
+open Pp
+open Lazy
+
+(** [get_type_of] performs beta reduction ;
+    Is it ok for Retyping.get_type_of (Zpower_nat n q) to return (fun _ : nat => Z) q ? *)
+let get_type_of env evd e =
+  Tacred.cbv_beta env evd (Retyping.get_type_of env evd e)
+
+(** [unsafe_to_constr c] returns a [Constr.t] without considering an evar_map.
+    This is useful for calling Constr.hash *)
+let unsafe_to_constr = EConstr.Unsafe.to_constr
+
+let pr_constr env evd e =
+  Printer.pr_constr_env env evd (unsafe_to_constr e)
+
+
+let debug = false (** Debug flag to get some intermediate terms *)
```

Comment:
Having to recompile to enable debug is a pain, either use Flags.debug or Goptions.declare_bool_option_and_ref.

Reply:
I know - I will suffer. 
I would rather remove the flag than expose it...
The printed information is currently only meaningful for chasing the last bug...

Reply:
> The printed information is currently only meaningful for chasing the last bug...

How do you know it is the last bug?

A debug infrastructure is often nice for future maintenance and debugging.

Reply:
I am just saying there is no debug infrastructure.

Reply:
> I am just saying there is no debug infrastructure.

Ah ok :)

Reply:
> I would rather remove the flag than expose it...

then remove it

Reply:
done.